internal/venice-common/src/main/java/com/linkedin/venice/utils/Utils.java

```diff
@@ -1141,4 +1142,16 @@ public static long parseDateTimeToEpoch(String dateTime, String dateTimeFormat,
     dateFormat.setTimeZone(TimeZone.getTimeZone(timeZone));
     return dateFormat.parse(dateTime).getTime();
   }
+
+  public static long getOSMemorySize() {
+    OperatingSystemMXBean osBean = ManagementFactory.getOperatingSystemMXBean();
+
+    if (osBean instanceof com.sun.management.OperatingSystemMXBean) {
+      com.sun.management.OperatingSystemMXBean extendedOsBean = (com.sun.management.OperatingSystemMXBean) osBean;
```

Comment:
Have you tested this function in Mac or Linux with OpenJDK?
If it doesn't work in these envs, this feature will be useless...

Reply:
Yes, this works in mac, and linux container/helix code uses same mechanism to fetch system memory.